### DISCUSSION TITLE
ESC Failsafe System

### DISCUSSION THREAD

**Discussion by datamann** [iNavFlight/inav#6103] [Ideas]
## Current Behavior
If the ESC current draw is two high or if the ESC temperature is to high, the FC has no way of knowing and often result in ESC failure or even worse fire.

## Desired Behavior
Make the FC sens high temp, or high current draw (ESC telemetry) and witch between to different ESC and even alert the pilot in video OSD and radio telem.

## Suggested Solution
Connect two ESCs to the FC or some kind of ESC module so that the FC can switch between two ESCs if one of the ESC fails.

## Who does this impact? Who is this for?
Most likely plane pilots

## Additional context
I suggested this to @DzikuVx in his live feed and was asked to register the feature here.

[a comment by an automated account is not shown]

**Comment by grizewald** [comment]
Why would the flight controller be better able to detect a problem with the ESC than the ESC itself?

Even if the flight controller could handle this,  being able to switch between two ESCs would require that the flight controller can also switch the main power from one ESC to the other, plus all the motor drive outputs. This is far from trivial to do and would add weight, complexity and additional failure points to the whole system.

It sounds to me like what you're asking for is a fault tolerant ESC. That's something for an ESC designer to create, not something for the flight controller to concern itself with.
